downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,554 @@
+
+```ballerina
+import ballerina/graphql;
+
+graphql:ListenerConfiguration configs = {
+    // http listener configurations
+};
+listener graphql:Listener graphqlListener = new (9090, configs);
+
+@graphql:ServiceConfigurration {maxQueryDepth: 3}
+ Service /graphql on graphqlListener {
+    // Service definition
+}
+```
+
+##### WebSub Module Improvements
+
+Included functionality to the `websub:SubscriberService` to respond with user-defined custom payloads/header parameters in error scenarios.
+
+```ballerina
+import ballerina/websub;
+
+listener websub:Listener subscriberListener = new (9001);
+
+service /subscriber on subscriberListener {
+    remote function onSubscriptionValidationDenied(websub:SubscriptionDeniedError msg) returns websub:Acknowledgement? {
+        websub:Acknowledgement ack = {
+            headers = {
+                        "Content-Encoding" : "gzip"
+                    },
+            body = {
+                     "message" : "Successfully processed request"
+                 }
+        };
+        return ack;
+    }
+
+    remote function onSubscriptionVerification(websub:SubscriptionVerification msg) returns 
+    websub:SubscriptionVerificationSuccess|websub:SubscriptionVerificationError {
+        if (msg.hubTopic == "https://www.sample.topic") {
+            return error websub:SubscriptionVerificationError("Hub topic not supported", headers = 
+            {"Content-Encoding": "gzip"}, body = {"message": "Hub topic not supported"});
+        } else {
+            return {};
+        }
+    }
+
+    remote function onEventNotification(websub:ContentDistributionMessage event) returns websub:Acknowledgement|
+    websub:SubscriptionDeletedError? {
+        return error websub:SubscriptionDeletedError("Subscriber wants to unsubscribe", headers = 
+        {"Content-Encoding": "gzip"}, body = {"message": "Unsubscribing from the topic"});
+    }
+}
+```
+
+##### WebSubHub Module Improvements
+
+Included functionality to the `websubhub:Service` to respond with user-defined custom payloads/header parameters in error scenarios.
+
+```ballerina
+import ballerina / websubhub;
+
+service /websubhub on websubhub:Listener(9091) {
+
+    remote function onRegisterTopic(websubhub:TopicRegistration message) returns websubhub:TopicRegistrationSuccess|
+    websubhub:TopicRegistrationError {
+        if (message.topic == "https://sub.topic.com") {
+            websubhub:TopicRegistrationSuccess successResult = {body: <map<string>>{isSuccess: "true"}};
+            return successResult;
+        } else {
+            return error websubhub:TopicRegistrationError("Topic registration failed!", headers = 
+            {"Content-Encoding": "gzip"}, body = {"hub.additional.details": "Feature is not supported in the hub"});
+        }
+    }
+
+}
```

Comment:
@MaryamZi here I have only provided one API method as a sample. There are 9 API methods to be implemented in total. Should we need to include all of them since this change is kind of common for all the APIs ?

Reply:
Then shall we change this as 
```suggestion
import ballerina/websubhub;

service /websubhub on new websubhub:Listener(9091) {
    remote function onRegisterTopic(websubhub:TopicRegistration message) 
            returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError {
        if (message.topic == "https://sub.topic.com") {
            websubhub:TopicRegistrationSuccess successResult = {body: <map<string>>{isSuccess: "true"}};
            return successResult;
        } else {
            return error websubhub:TopicRegistrationError("Topic registration failed!", 
                        headers = {"Content-Encoding": "gzip"}, 
                        body = {"hub.additional.details": "Feature is not supported in the hub"});
        }
    }

    // Other remote methods...
}
```

Reply:
Yes this is okay